honu_test.go

```diff
@@ -34,113 +42,123 @@ func setupHonuDB() (*honu.DB, string, error) {
 		GossipSigma:    15 * time.Second,
 	}
 
-	db, err := honu.Open(uri, conf)
-	return db, tmpDir, err
-}
-
-func setupLevelDB() (*leveldb.DB, string, error) {
-	// Create a new leveldb database in a temporary directory
-	tmpDir, err := ioutil.TempDir("", "honuldb-*")
-	if err != nil {
-		return nil, "", err
-	}
-
-	// Open a leveldb database directly without honu wrapper
-	db, err := leveldb.OpenFile(tmpDir, nil)
-	return db, tmpDir, err
-}
-
-func TestLevelDBInteractions(t *testing.T) {
-	db, tmpDir, err := setupHonuDB()
+	db, err = honu.Open(uri, conf)
+	require.NoError(t, err)
 	if err != nil && tmpDir != "" {
 		fmt.Println(tmpDir)
 		os.RemoveAll(tmpDir)
 	}
-	require.NoError(t, err)
+	return db, tmpDir
+}
+
+func TestLevelDBInteractions(t *testing.T) {
+	db, tmpDir := setupHonuDB(t)
 
 	// Cleanup when we're done with the test
 	defer os.RemoveAll(tmpDir)
 	defer db.Close()
 
-	// Put a version to the database
-	_, err = db.Put([]byte("foo"), []byte("this is the value of foo"))
-	require.NoError(t, err)
-
-	// Get the version of foo from the database
-	value, err := db.Get([]byte("foo"))
-	require.NoError(t, err)
-	require.Equal(t, []byte("this is the value of foo"), value)
+	for _, namespace := range getNamespaces() {
+		// Use a constant key to ensure namespaces
+		// are working correctly.
+		key := []byte("foo")
+		//append a constant to namespace as the value
+		//because when the empty namespace is returned
+		//as a key it is unmarsheled as []byte(nil)
+		//instead of []byte{}
+		expectedValue := []byte(namespace + "this is the value of foo")
```

Comment:
I should have made that comment clearer, the primary reason I'm using the namespace as the value rather than using a constant is to ensure that the correct value is returned from Get() for each separate namespace. I was appending a constant rather than just using the namespace as the value because I was having issues with the blank ("") namespace.